pandas/core/indexes/datetimes.py

```diff
@@ -2137,11 +2137,30 @@ def _generate_regular_range(start, end, periods, offset):
             tz = start.tz
         elif start is not None:
             b = Timestamp(start).value
-            e = b + np.int64(periods) * stride
+            try:
+                with np.errstate(over='raise'):
+                    # raise instead of incorrectly wrapping around GH#19740
+                    e = b + np.int64(periods) * stride
+            except (FloatingPointError, OverflowError):
+                raise libts.OutOfBoundsDatetime('Cannot generate range with '
```

Comment:
there is already a routine to do this checking (not called here) pls use it

Reply:
greo for overflow

Reply:
OK, I'll track that down (grep turns up a lot of stackoverflow...).  In the interim, #19582, #19611, #19650, #19659 , #19661 have been reviewed, should be straightforward to finish off.  Test refactoring in #19736, #19739 should be easy but haven't been reviewed.

Reply:
grep the codebase (I put it below)